### PR TITLE
Add ESG

### PR DESCRIPTION
**Pull request by xinyuezhao**
- add fv_e_sg.go
- add fvESg_service.go

### CODE REVIEW

File: client/fvESg_service.go [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+	list := models.EndpointSecurityGroupListFromContainer(cont)
+	return list, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsSecInheritedFromEndpointSecurityGroup( parentDn, tDn string) error {
+	dn := fmt.Sprintf("%s/rssecInherited-[%s]", parentDn, tDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"
```
Comment (AshuSoni-crest) [inline client/fvESg_service.go:55]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/fvESg_service.go [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+	}
+	return st, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsProvFromEndpointSecurityGroup( parentDn, tnVzBrCPName string) error {
+	dn := fmt.Sprintf("%s/rsprov-%s", parentDn, tnVzBrCPName)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"
```
Comment (AshuSoni-crest) [inline client/fvESg_service.go:99]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/fvESg_service.go [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+	}
+	return st, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsConsIfFromEndpointSecurityGroup( parentDn, tnVzCPIfName string) error {
+	dn := fmt.Sprintf("%s/rsconsIf-%s", parentDn, tnVzCPIfName)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"
```
Comment (AshuSoni-crest) [inline client/fvESg_service.go:143]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/fvESg_service.go [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+	}
+	return st, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsCustQosPolFromEndpointSecurityGroup( parentDn, tnQosCustomPolName string) error {
+	dn := fmt.Sprintf("%s/rscustQosPol", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s","tnQosCustomPolName": "%s"
```
Comment (AshuSoni-crest) [inline client/fvESg_service.go:187]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/fvESg_service.go [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+		return nil,err
+	}
+}
+
+func (sm *ServiceManager) CreateRelationfvRsConsFromEndpointSecurityGroup( parentDn, tnVzBrCPName string) error {
+	dn := fmt.Sprintf("%s/rscons-%s", parentDn, tnVzBrCPName)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"
```
Comment (AshuSoni-crest) [inline client/fvESg_service.go:225]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/fvESg_service.go [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+	}
+	return st, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsScopeFromEndpointSecurityGroup( parentDn, tnFvCtxName string) error {
+	dn := fmt.Sprintf("%s/rsscope", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s","tnFvCtxName": "%s"
```
Comment (AshuSoni-crest) [inline client/fvESg_service.go:269]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/fvESg_service.go [hunk 7]
```diff
@@ -0,0 +1,389 @@
...
+		return nil,err
+	}
+}
+
+func (sm *ServiceManager) CreateRelationfvRsProtByFromEndpointSecurityGroup( parentDn, tnVzTabooName string) error {
+	dn := fmt.Sprintf("%s/rsprotBy-%s", parentDn, tnVzTabooName)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"
```
Comment (AshuSoni-crest) [inline client/fvESg_service.go:307]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/fvESg_service.go [hunk 8]
```diff
@@ -0,0 +1,389 @@
...
+	}
+	return st, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsIntraEpgFromEndpointSecurityGroup( parentDn, tnVzBrCPName string) error {
+	dn := fmt.Sprintf("%s/rsintraEpg-%s", parentDn, tnVzBrCPName)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"
```
Comment (AshuSoni-crest) [inline client/fvESg_service.go:351]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

**Review by AshuSoni-crest — CHANGES_REQUESTED** [review]
some changes needed. Added comments in files

**Review by lhercot — APPROVED** [review]

**Pull request merged**